Engine/src/Engine/UserInterface.h

```diff
@@ -0,0 +1,59 @@
+#pragma once
+
+#include <map>
+#include <string>
+
+#include "imgui.h"
+#include "imgui_impl_opengl3.h"
+#include "imgui_impl_glfw.h"
+
+#include "glm/glm.hpp"
+
+class UserInterface
+{
+
+public:
+
+	//initializes ImGUI vec4s to be used for colours
+	static ImVec4 red;
+	static ImVec4 green;
+	static ImVec4 blue;
+	static ImVec4 white;
+
+	//initialize start and end functions
+	static void startUI();
+	static void endUI();
+
+	//loadFont takes in a string file name, float font size, and string tag and returns true if the font is valid
+	static bool loadFont(const std::string& fontFilename, float fontSize, const std::string& tag);
+
+	virtual ~UserInterface() = 0 {}
```

Comment:
changing `virtual ~UserInterface() = 0 {}` to `virtual ~UserInterface() = 0;` caused the build to fail. Not sure why.

Reply:
I guess it's because the destructor for UserInterface still needs to be defined since this is being called. Usually " = 0 " at the end of a function indicates that it is a pure virtual function, meaning that the derived classes HAVE to implement the function. Declaring a body in the same base class ignores that functionality. I think in that case we can just use the default constructor (or in our case to follow the rule of 0, just get rid of it altogether).